src/github.com/matrix-org/dendrite/clientapi/routing/joinroom.go

```diff
@@ -65,7 +70,18 @@ func JoinRoomByIDOrAlias(
 	content["displayname"] = profile.DisplayName
 	content["avatar_url"] = profile.AvatarURL
 
-	r := joinRoomReq{req, content, device.UserID, cfg, federation, producer, queryAPI, aliasAPI, keyRing}
+	r := joinRoomReq{
```

Comment:
I would've preferred the single/double line initialisation, since
1) this is a private struct.
2) Field names are more or less exactly the same as variable names.
But I'll leave it upto you.

Reply:
done

Reply:
Thanks :)